### ISSUE TITLE
DateTimeExpander's boolean calendar features turn a missing date into False instead of missing

### ISSUE THREAD

**Issue by bijay-odyssey** [bijay-odyssey/edaprep#41]
`DateTimeExpander`'s numeric calendar features correctly propagate a missing date as a
missing value: `d__dayofweek` is `NaN` for a `NaT` row. Its boolean features don't.
`is_weekend` is derived as `(dayofweek >= 5)`, and pandas' comparison of `NaN >= 5`
evaluates to `False`, not `NaN` -- so a row whose date is unknown is reported as
"definitely not a weekend" instead of "unknown". The other boolean extractors
(`is_month_start`, `is_month_end`, `is_quarter_start`, `is_quarter_end`,
`is_year_start`, `is_year_end`, which go through pandas' `.dt.is_*` accessors) have the
same property: `NaT.is_month_start` is `False`, not `NaN`.

### Reproduced

```python
df = pd.DataFrame({"d": pd.to_datetime(["2024-01-01", None, "2024-03-01", "2024-01-06"])})
DateTimeExpander(["d"]).fit_transform(df, None, ctx)[["d__dayofweek", "d__is_weekend"]]
#    d__dayofweek  d__is_weekend
# 0           0.0            0.0
# 1           NaN            0.0   <- row 1's date is unknown; is_weekend is fabricated
# 2           4.0            0.0
# 3           5.0            1.0
```

Row 1 has no date at all, yet `d__is_weekend` reports `0.0` (false) with the same
confidence as row 0, whose date really is a Monday. Nothing is counted or reported
for this -- it's a silently fabricated value, not a flagged one.

### What to do

The extractors live in `_FEATURE_FUNCS`, a `{name: lambda s: ...}` dict
(`datetime_features.py:41-56`). Each boolean one currently does
`(...).astype("float64")` with no missingness handling, e.g.:

```python
"is_weekend": lambda s: (s.dt.dayofweek >= 5).astype("float64"),
"is_month_start": lambda s: s.dt.is_month_start.astype("float64"),
```

Mask each to `NaN` wherever the source is `NaT`, e.g.
`lambda s: (s.dt.dayofweek >= 5).astype("float64").where(s.notna())`, applied to all
seven boolean entries (`is_weekend`, `is_month_start`, `is_month_end`,
`is_quarter_start`, `is_quarter_end`, `is_year_start`, `is_year_end`). Simplest to do
in one place rather than seven if there's a natural point after the lambda is called
where `s` (the original series) is still in scope to build the mask from.

### Tests

A frame with at least one `NaT` date, asserting every boolean calendar feature is
`NaN` (not `0.0`/`False`) for that row, alongside the existing numeric-feature
missingness behaviour.

**Before you start:** please comment to claim this issue, and check whether someone
already has.
